lib/Dispatcher.js

```diff
@@ -39,29 +47,35 @@ module.exports = function () {
      * @throws {Error} if store is invalid
      * @throws {Error} if store is already registered
      */
-    Dispatcher.registerStore = function registerStore(store) {
-        if ('function' !== typeof store) {
+    registerStore: function registerStore(store) {
+        if (!BaseStore.isPrototypeOf(store)) {
             throw new Error('registerStore requires a constructor as first parameter');
         }
-        var storeName = Dispatcher.getStoreName(store);
+
+        var storeName = this.getStoreName(store);
         if (!storeName) {
             throw new Error('Store is required to have a `storeName` property.');
         }
-        if (Dispatcher.stores[storeName]) {
-            if (Dispatcher.stores[storeName] === store) {
+
+        if (!this.hasOwnProperty('stores')) {
+            this.stores = {};
```

Comment:
this is how we avoid leaking to the on the parent stores, if the current store don't has it own stores instance it will receive a clean object before registering a new store.